unreal_asset/src/lib.rs

```diff
@@ -245,20 +247,22 @@ impl<'asset, 'cursor> AssetSerializer<'asset, 'cursor> {
     }
 }
 
-impl<'asset, 'cursor> AssetTrait for AssetSerializer<'asset, 'cursor> {
+impl<'asset, 'cursor, W: Read + Seek + Write, C: Read + Seek> AssetTrait
+    for AssetSerializer<'asset, 'cursor, W, C>
+{
     fn get_custom_version<T>(&self) -> CustomVersion
     where
         T: CustomVersionTrait + Into<i32>,
     {
         self.asset.get_custom_version::<T>()
     }
 
-    fn position(&self) -> u64 {
-        self.cursor.position()
+    fn position(&mut self) -> u64 {
```

Comment:
This operation is not logically mutating and should therefore not require a &mut.

Reply:
I can't think of an alternative that keeps it generic because only cursors store position and it's not harmful since it's only part of the internal API. Besides, all it's doing is self.seek(io::SeekFrom::Current(0))

Reply:
Makes sense